### PR TITLE
fix: Azure SEV-SNP version always gets overwritten by latest API versions

### PR DESCRIPTION
**Pull request by elchead**
We found this bug earlier and fixed it but somehow it appeared again.

**Why?**
When the user sets an attestation version, the value gets overwritten by the latest value. 

### Proposed change(s)

- check if latest is set before setting latest numerical value

### Checklist

- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Link to Milestone

### CODE REVIEW

File: internal/config/config_test.go [hunk 1]
```diff
@@ -77,17 +78,68 @@ func TestReadConfigFile(t *testing.T) {
...
+			assert.Equal(tc.wantResult, result)
+		})
+	}
+}
+
+func modifyConfigForAzureToPassValidate(c *Config) {
+	c.RemoveProviderAndAttestationExcept(cloudprovider.Azure)
+	c.Image = "v" + constants.VersionInfo()
+	c.Provider.Azure.SubscriptionID = "f4278079-288c-4766-a98c-ab9d5dba01a5"
+	c.Provider.Azure.TenantID = "d4ff9d63-6d6d-4042-8f6a-21e804add5aa"
```
Comment (derpsteb) [inline internal/config/config_test.go:126]: can we use values like `11111111-1111-1111-1111-111111111111` here?

File: internal/config/azure.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	if err != nil {
+		return err
+	}
+	// set number and keep isLatest flag
+	c.mergeWithLatestVersion(versions.AzureSEVSNPVersion)
+	return nil
+}
+
+func (c *AzureSEVSNP) mergeWithLatestVersion(latest attestationconfigapi.AzureSEVSNPVersion) {
+	if c.BootloaderVersion.IsLatest {
```
Comment (derpsteb) [inline internal/config/azure.go:84]: The `IsLatest` name seems a bit counterintuitive to me. At least in this place. `WantLatest` would represent the intention of the variable more accurately, wouldn't it?

Comment (elchead) [reply internal/config/azure.go:84]: Want is better,  I agree. naming is hard :)

Comment (derpsteb) [reply internal/config/azure.go:84]: it is! :D

**Review by derpsteb — APPROVED** [review]
lgtm :+1:

**Pull request merged**